src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -114,11 +125,25 @@ private void Initialize(Compilation comp, AnalyzerExecutor analyzerExecutor, Can
             }
         }
 
-        private Task ExecuteSyntaxTreeActions(CancellationToken cancellationToken)
+        public void StartCompleteAnalysis(ImmutableHashSet<SyntaxTree> syntaxTreesToSkip, CancellationToken cancellationToken)
+        {
+            if (_initializeTaskStarted && _primaryTask == null)
+            {
+                _primaryTask = Task.Run(async () =>
```

Comment:
Can you please add a comment to this method that the caller needs to ensure thread safety?